lms/services/canvas_studio.py

```diff
@@ -316,6 +403,56 @@ def _api_url(self, path: str) -> str:
         site = self._canvas_studio_site()
         return f"{site}/api/public/{path}"
 
+    def _admin_email(self) -> str:
+        """Return the email address of the configured Canvas Studio admin."""
+        admin_email = self._application_instance.settings.get(
+            "canvas_studio", "admin_email"
+        )
+        if not admin_email:
+            raise HTTPBadRequest(
+                "Admin account is not configured for Canvas Studio integration"
+            )
+        return admin_email
+
+    def _is_admin(self) -> bool:
+        """Return true if the current LTI user is the configure Canvas Studio admin."""
+        return self._request.lti_user.email == self._admin_email()
+
+    @property
+    @lru_cache
+    def _admin_oauth_http(self) -> OAuthHTTPService:
+        """
+        Return an OAuthHTTPService that makes calls using the admin user account.
+
+        Admin accounts have the ability to download all videos and transcripts
+        in a Canvas Studio instance, whereas videos can ordinarily only be
+        downloaded by the owner. Therefore when launching Canvas Studio
+        assignments, we use this account instead of the current user's account
+        to authenticate the API requests for downloading videos and transcripts.
+        """
+
+        # The caller should check for this condition before calling this method
+        # and use the standard `self._oauth_http_service` property instead.
+        assert not self._is_admin()
+
+        admin_email = self._admin_email()
+        admin_user = (
```

Comment:
You can do one_or_none in a search type interface (but it will be uglier) but you can't do search in a fetch/lookup interface.

One alternative we haven't used yet would be to expose the search query method returning a query object instead of one or a list of models.

That way the centralization benedict of having it over UserService will still be there (eg I we wanted to exclude an hypothetical `delted=True` it will be a one line change) but you still get the flexibility of other services executing the query themselves (ie they can use .one() .one_or_none() or even use the query as a subquery...)